src/systemctl/systemctl-enable.c

```diff
@@ -11,46 +11,51 @@
 #include "systemctl-util.h"
 #include "systemctl.h"
 
-static int normalize_filenames(char **names) {
-        int r;
+static int normalize_link_paths(char **paths) {
 
-        STRV_FOREACH(u, names)
-                if (!path_is_absolute(*u)) {
-                        char* normalized_path;
+        STRV_FOREACH(u, paths) {
+                if (path_is_absolute(*u))
+                        continue;
 
-                        if (!isempty(arg_root))
-                                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
-                                                       "Non-absolute paths are not allowed when --root is used: %s",
-                                                       *u);
+                if (!isempty(arg_root))
+                        return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                               "Non-absolute paths are not allowed when --root= is used: %s",
+                                               *u);
 
-                        if (!strchr(*u, '/'))
-                                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
-                                                       "Link argument must contain at least one directory separator.\n"
-                                                       "If you intended to link a file in the current directory, try ./%s instead.",
-                                                       *u);
+                if (!is_path(*u))
+                        return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
+                                               "Link argument must contain at least one directory separator.\n"
+                                               "If you intended to link a file in the current directory, try './%s' instead.",
+                                               *u);
 
-                        r = path_make_absolute_cwd(*u, &normalized_path);
-                        if (r < 0)
-                                return r;
+                char *normalized_path;
 
-                        free_and_replace(*u, normalized_path);
-                }
+                if (path_make_absolute_cwd(*u, &normalized_path) < 0)
```

Comment:
path_make_absolute_cwd() can actually fail for more reasons than oom. i.e. if the cwd is not accesisble anymore due to nfs shit, or because of namespace shenanigans and stuff. I'd prefer if we write a proper log messag ehere in cas of failure instead of just log_oom()